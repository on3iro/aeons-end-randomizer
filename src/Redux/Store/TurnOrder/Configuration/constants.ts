import AERData from 'aer-data/src/index'

const INITIAL_MODE = 'Default'
const INITIAL_PLAYER_SETUP = AERData.turnordersetups['onePlayer']
const INITIAL_VARIATION = INITIAL_PLAYER_SETUP.variations['default']

const TURNORDER_CONFIG_DB_KEY = '2-4-turnOrderConfiguration'

export {
  INITIAL_MODE,
  INITIAL_PLAYER_SETUP,
  INITIAL_VARIATION,
  TURNORDER_CONFIG_DB_KEY,
}
