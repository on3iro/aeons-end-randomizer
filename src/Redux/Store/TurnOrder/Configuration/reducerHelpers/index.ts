export * from './setMode'
export * from './selectPlayerCount'
export * from './selectSetup'
export * from './fetchFromDb'
