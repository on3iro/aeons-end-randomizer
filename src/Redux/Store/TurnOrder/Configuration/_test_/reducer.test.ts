import { ActionTypes } from 'Redux/Store/TurnOrder/Configuration/types'
import {
  initialState,
  Reducer,
} from 'Redux/Store/TurnOrder/Configuration/reducer'

describe('Turnorder configuration reducer', () => {
  it('should return the initial state', () => {
    // @ts-ignore
    expect(Reducer(undefined, {})).toMatchSnapshot()
  })

  it('should handle SET_MODE', () => {
    expect(
      Reducer(initialState, {
        type: ActionTypes.SET_MODE,
        payload: 'Blitz',
      })
    ).toMatchSnapshot()
  })

  it('should handle SELECT_PLAYER_COUNT', () => {
    expect(
      Reducer(initialState, {
        type: ActionTypes.SELECT_PLAYER_COUNT,
        payload: 'onePlayer',
      })
    ).toMatchSnapshot()
  })

  it('should handle SELECT_SETUP', () => {
    expect(
      Reducer(initialState, {
        type: ActionTypes.SELECT_SETUP,
        payload: 'default',
      })
    ).toMatchSnapshot()
  })

  it('should handle FETCH_FROM_DB', () => {
    expect(
      Reducer(initialState, {
        type: ActionTypes.FETCH_FROM_DB,
      })
    ).toMatchSnapshot()
  })

  it('should handle FETCH_FROM_DB_SUCCESS', () => {
    expect(
      Reducer(initialState, {
        type: ActionTypes.FETCH_FROM_DB_SUCCESS,
        payload: {
          ...initialState,
          Mode: 'Maelstrom',
        },
      })
    ).toMatchSnapshot()
  })

  it('should handle FETCH_FROM_DB_FAILURE', () => {
    expect(
      Reducer(initialState, {
        type: ActionTypes.FETCH_FROM_DB_FAILURE,
      })
    ).toMatchSnapshot()
  })
})
