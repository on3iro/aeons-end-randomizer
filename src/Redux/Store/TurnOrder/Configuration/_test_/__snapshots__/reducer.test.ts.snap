// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Turnorder configuration reducer should handle FETCH_FROM_DB 1`] = `
Array [
  Object {
    "Mode": "Default",
    "SelectedPlayerCount": Object {
      "id": "onePlayer",
      "name": "1",
      "variations": Object {
        "34alternating": Object {
          "id": "34alternating",
          "name": "3/4 Cards alternating",
          "turnOrderCards": Array [
            Object {
              "id": "player1-1",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-2",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-3",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "alternate": true,
              "id": "player-1-alternating",
              "name": "Player 1 (alternate)",
              "type": "player1",
            },
            Object {
              "id": "nemesis-1",
              "name": "Nemesis",
              "type": "nemesis",
            },
            Object {
              "id": "nemesis-2",
              "name": "Nemesis",
              "type": "nemesis",
            },
          ],
        },
        "4toc": Object {
          "id": "4toc",
          "name": "4 ToC",
          "turnOrderCards": Array [
            Object {
              "id": "player1-1",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-2",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-3",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-4",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "nemesis-1",
              "name": "Nemesis",
              "type": "nemesis",
            },
            Object {
              "id": "nemesis-2",
              "name": "Nemesis",
              "type": "nemesis",
            },
          ],
        },
        "default": Object {
          "id": "default",
          "name": "Default (3 ToC)",
          "turnOrderCards": Array [
            Object {
              "id": "player1-1",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-2",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-3",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "nemesis-1",
              "name": "Nemesis",
              "type": "nemesis",
            },
            Object {
              "id": "nemesis-2",
              "name": "Nemesis",
              "type": "nemesis",
            },
          ],
        },
      },
    },
    "SelectedSetup": Object {
      "id": "default",
      "name": "Default (3 ToC)",
      "turnOrderCards": Array [
        Object {
          "id": "player1-1",
          "name": "Player 1",
          "type": "player1",
        },
        Object {
          "id": "player1-2",
          "name": "Player 1",
          "type": "player1",
        },
        Object {
          "id": "player1-3",
          "name": "Player 1",
          "type": "player1",
        },
        Object {
          "id": "nemesis-1",
          "name": "Nemesis",
          "type": "nemesis",
        },
        Object {
          "id": "nemesis-2",
          "name": "Nemesis",
          "type": "nemesis",
        },
      ],
    },
  },
  Object {
    "args": Array [
      "2-4-turnOrderConfiguration",
    ],
    "failActionCreator": [Function],
    "func": [Function],
    "simulate": [Function],
    "successActionCreator": [Function],
    "type": "RUN",
    Symbol(isCmd): true,
  },
]
`;

exports[`Turnorder configuration reducer should handle FETCH_FROM_DB_FAILURE 1`] = `
Object {
  "Mode": "Default",
  "SelectedPlayerCount": Object {
    "id": "onePlayer",
    "name": "1",
    "variations": Object {
      "34alternating": Object {
        "id": "34alternating",
        "name": "3/4 Cards alternating",
        "turnOrderCards": Array [
          Object {
            "id": "player1-1",
            "name": "Player 1",
            "type": "player1",
          },
          Object {
            "id": "player1-2",
            "name": "Player 1",
            "type": "player1",
          },
          Object {
            "id": "player1-3",
            "name": "Player 1",
            "type": "player1",
          },
          Object {
            "alternate": true,
            "id": "player-1-alternating",
            "name": "Player 1 (alternate)",
            "type": "player1",
          },
          Object {
            "id": "nemesis-1",
            "name": "Nemesis",
            "type": "nemesis",
          },
          Object {
            "id": "nemesis-2",
            "name": "Nemesis",
            "type": "nemesis",
          },
        ],
      },
      "4toc": Object {
        "id": "4toc",
        "name": "4 ToC",
        "turnOrderCards": Array [
          Object {
            "id": "player1-1",
            "name": "Player 1",
            "type": "player1",
          },
          Object {
            "id": "player1-2",
            "name": "Player 1",
            "type": "player1",
          },
          Object {
            "id": "player1-3",
            "name": "Player 1",
            "type": "player1",
          },
          Object {
            "id": "player1-4",
            "name": "Player 1",
            "type": "player1",
          },
          Object {
            "id": "nemesis-1",
            "name": "Nemesis",
            "type": "nemesis",
          },
          Object {
            "id": "nemesis-2",
            "name": "Nemesis",
            "type": "nemesis",
          },
        ],
      },
      "default": Object {
        "id": "default",
        "name": "Default (3 ToC)",
        "turnOrderCards": Array [
          Object {
            "id": "player1-1",
            "name": "Player 1",
            "type": "player1",
          },
          Object {
            "id": "player1-2",
            "name": "Player 1",
            "type": "player1",
          },
          Object {
            "id": "player1-3",
            "name": "Player 1",
            "type": "player1",
          },
          Object {
            "id": "nemesis-1",
            "name": "Nemesis",
            "type": "nemesis",
          },
          Object {
            "id": "nemesis-2",
            "name": "Nemesis",
            "type": "nemesis",
          },
        ],
      },
    },
  },
  "SelectedSetup": Object {
    "id": "default",
    "name": "Default (3 ToC)",
    "turnOrderCards": Array [
      Object {
        "id": "player1-1",
        "name": "Player 1",
        "type": "player1",
      },
      Object {
        "id": "player1-2",
        "name": "Player 1",
        "type": "player1",
      },
      Object {
        "id": "player1-3",
        "name": "Player 1",
        "type": "player1",
      },
      Object {
        "id": "nemesis-1",
        "name": "Nemesis",
        "type": "nemesis",
      },
      Object {
        "id": "nemesis-2",
        "name": "Nemesis",
        "type": "nemesis",
      },
    ],
  },
}
`;

exports[`Turnorder configuration reducer should handle FETCH_FROM_DB_SUCCESS 1`] = `
Array [
  Object {
    "Mode": "Maelstrom",
    "SelectedPlayerCount": Object {
      "id": "onePlayer",
      "name": "1",
      "variations": Object {
        "34alternating": Object {
          "id": "34alternating",
          "name": "3/4 Cards alternating",
          "turnOrderCards": Array [
            Object {
              "id": "player1-1",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-2",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-3",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "alternate": true,
              "id": "player-1-alternating",
              "name": "Player 1 (alternate)",
              "type": "player1",
            },
            Object {
              "id": "nemesis-1",
              "name": "Nemesis",
              "type": "nemesis",
            },
            Object {
              "id": "nemesis-2",
              "name": "Nemesis",
              "type": "nemesis",
            },
          ],
        },
        "4toc": Object {
          "id": "4toc",
          "name": "4 ToC",
          "turnOrderCards": Array [
            Object {
              "id": "player1-1",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-2",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-3",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-4",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "nemesis-1",
              "name": "Nemesis",
              "type": "nemesis",
            },
            Object {
              "id": "nemesis-2",
              "name": "Nemesis",
              "type": "nemesis",
            },
          ],
        },
        "default": Object {
          "id": "default",
          "name": "Default (3 ToC)",
          "turnOrderCards": Array [
            Object {
              "id": "player1-1",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-2",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-3",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "nemesis-1",
              "name": "Nemesis",
              "type": "nemesis",
            },
            Object {
              "id": "nemesis-2",
              "name": "Nemesis",
              "type": "nemesis",
            },
          ],
        },
      },
    },
    "SelectedSetup": Object {
      "id": "default",
      "name": "Default (3 ToC)",
      "turnOrderCards": Array [
        Object {
          "id": "player1-1",
          "name": "Player 1",
          "type": "player1",
        },
        Object {
          "id": "player1-2",
          "name": "Player 1",
          "type": "player1",
        },
        Object {
          "id": "player1-3",
          "name": "Player 1",
          "type": "player1",
        },
        Object {
          "id": "nemesis-1",
          "name": "Nemesis",
          "type": "nemesis",
        },
        Object {
          "id": "nemesis-2",
          "name": "Nemesis",
          "type": "nemesis",
        },
      ],
    },
  },
  Object {
    "actionToDispatch": Object {
      "type": "TurnOrder/ActiveGame/FETCH_FROM_DB",
    },
    "simulate": [Function],
    "type": "ACTION",
    Symbol(isCmd): true,
  },
]
`;

exports[`Turnorder configuration reducer should handle SELECT_PLAYER_COUNT 1`] = `
Array [
  Object {
    "Mode": "Default",
    "SelectedPlayerCount": Object {
      "id": "onePlayer",
      "name": "1",
      "variations": Object {
        "34alternating": Object {
          "id": "34alternating",
          "name": "3/4 Cards alternating",
          "turnOrderCards": Array [
            Object {
              "id": "player1-1",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-2",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-3",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "alternate": true,
              "id": "player-1-alternating",
              "name": "Player 1 (alternate)",
              "type": "player1",
            },
            Object {
              "id": "nemesis-1",
              "name": "Nemesis",
              "type": "nemesis",
            },
            Object {
              "id": "nemesis-2",
              "name": "Nemesis",
              "type": "nemesis",
            },
          ],
        },
        "4toc": Object {
          "id": "4toc",
          "name": "4 ToC",
          "turnOrderCards": Array [
            Object {
              "id": "player1-1",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-2",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-3",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-4",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "nemesis-1",
              "name": "Nemesis",
              "type": "nemesis",
            },
            Object {
              "id": "nemesis-2",
              "name": "Nemesis",
              "type": "nemesis",
            },
          ],
        },
        "default": Object {
          "id": "default",
          "name": "Default (3 ToC)",
          "turnOrderCards": Array [
            Object {
              "id": "player1-1",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-2",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-3",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "nemesis-1",
              "name": "Nemesis",
              "type": "nemesis",
            },
            Object {
              "id": "nemesis-2",
              "name": "Nemesis",
              "type": "nemesis",
            },
          ],
        },
      },
    },
    "SelectedSetup": Object {
      "id": "default",
      "name": "Default (3 ToC)",
      "turnOrderCards": Array [
        Object {
          "id": "player1-1",
          "name": "Player 1",
          "type": "player1",
        },
        Object {
          "id": "player1-2",
          "name": "Player 1",
          "type": "player1",
        },
        Object {
          "id": "player1-3",
          "name": "Player 1",
          "type": "player1",
        },
        Object {
          "id": "nemesis-1",
          "name": "Nemesis",
          "type": "nemesis",
        },
        Object {
          "id": "nemesis-2",
          "name": "Nemesis",
          "type": "nemesis",
        },
      ],
    },
  },
  Object {
    "args": Array [
      "2-4-turnOrderConfiguration",
      Object {
        "Mode": "Default",
        "SelectedPlayerCount": Object {
          "id": "onePlayer",
          "name": "1",
          "variations": Object {
            "34alternating": Object {
              "id": "34alternating",
              "name": "3/4 Cards alternating",
              "turnOrderCards": Array [
                Object {
                  "id": "player1-1",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "player1-2",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "player1-3",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "alternate": true,
                  "id": "player-1-alternating",
                  "name": "Player 1 (alternate)",
                  "type": "player1",
                },
                Object {
                  "id": "nemesis-1",
                  "name": "Nemesis",
                  "type": "nemesis",
                },
                Object {
                  "id": "nemesis-2",
                  "name": "Nemesis",
                  "type": "nemesis",
                },
              ],
            },
            "4toc": Object {
              "id": "4toc",
              "name": "4 ToC",
              "turnOrderCards": Array [
                Object {
                  "id": "player1-1",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "player1-2",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "player1-3",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "player1-4",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "nemesis-1",
                  "name": "Nemesis",
                  "type": "nemesis",
                },
                Object {
                  "id": "nemesis-2",
                  "name": "Nemesis",
                  "type": "nemesis",
                },
              ],
            },
            "default": Object {
              "id": "default",
              "name": "Default (3 ToC)",
              "turnOrderCards": Array [
                Object {
                  "id": "player1-1",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "player1-2",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "player1-3",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "nemesis-1",
                  "name": "Nemesis",
                  "type": "nemesis",
                },
                Object {
                  "id": "nemesis-2",
                  "name": "Nemesis",
                  "type": "nemesis",
                },
              ],
            },
          },
        },
        "SelectedSetup": Object {
          "id": "default",
          "name": "Default (3 ToC)",
          "turnOrderCards": Array [
            Object {
              "id": "player1-1",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-2",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-3",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "nemesis-1",
              "name": "Nemesis",
              "type": "nemesis",
            },
            Object {
              "id": "nemesis-2",
              "name": "Nemesis",
              "type": "nemesis",
            },
          ],
        },
      },
    ],
    "failActionCreator": [Function],
    "func": [Function],
    "simulate": [Function],
    "successActionCreator": [Function],
    "type": "RUN",
    Symbol(isCmd): true,
  },
]
`;

exports[`Turnorder configuration reducer should handle SELECT_SETUP 1`] = `
Array [
  Object {
    "Mode": "Default",
    "SelectedPlayerCount": Object {
      "id": "onePlayer",
      "name": "1",
      "variations": Object {
        "34alternating": Object {
          "id": "34alternating",
          "name": "3/4 Cards alternating",
          "turnOrderCards": Array [
            Object {
              "id": "player1-1",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-2",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-3",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "alternate": true,
              "id": "player-1-alternating",
              "name": "Player 1 (alternate)",
              "type": "player1",
            },
            Object {
              "id": "nemesis-1",
              "name": "Nemesis",
              "type": "nemesis",
            },
            Object {
              "id": "nemesis-2",
              "name": "Nemesis",
              "type": "nemesis",
            },
          ],
        },
        "4toc": Object {
          "id": "4toc",
          "name": "4 ToC",
          "turnOrderCards": Array [
            Object {
              "id": "player1-1",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-2",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-3",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-4",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "nemesis-1",
              "name": "Nemesis",
              "type": "nemesis",
            },
            Object {
              "id": "nemesis-2",
              "name": "Nemesis",
              "type": "nemesis",
            },
          ],
        },
        "default": Object {
          "id": "default",
          "name": "Default (3 ToC)",
          "turnOrderCards": Array [
            Object {
              "id": "player1-1",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-2",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-3",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "nemesis-1",
              "name": "Nemesis",
              "type": "nemesis",
            },
            Object {
              "id": "nemesis-2",
              "name": "Nemesis",
              "type": "nemesis",
            },
          ],
        },
      },
    },
    "SelectedSetup": Object {
      "id": "default",
      "name": "Default (3 ToC)",
      "turnOrderCards": Array [
        Object {
          "id": "player1-1",
          "name": "Player 1",
          "type": "player1",
        },
        Object {
          "id": "player1-2",
          "name": "Player 1",
          "type": "player1",
        },
        Object {
          "id": "player1-3",
          "name": "Player 1",
          "type": "player1",
        },
        Object {
          "id": "nemesis-1",
          "name": "Nemesis",
          "type": "nemesis",
        },
        Object {
          "id": "nemesis-2",
          "name": "Nemesis",
          "type": "nemesis",
        },
      ],
    },
  },
  Object {
    "args": Array [
      "2-4-turnOrderConfiguration",
      Object {
        "Mode": "Default",
        "SelectedPlayerCount": Object {
          "id": "onePlayer",
          "name": "1",
          "variations": Object {
            "34alternating": Object {
              "id": "34alternating",
              "name": "3/4 Cards alternating",
              "turnOrderCards": Array [
                Object {
                  "id": "player1-1",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "player1-2",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "player1-3",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "alternate": true,
                  "id": "player-1-alternating",
                  "name": "Player 1 (alternate)",
                  "type": "player1",
                },
                Object {
                  "id": "nemesis-1",
                  "name": "Nemesis",
                  "type": "nemesis",
                },
                Object {
                  "id": "nemesis-2",
                  "name": "Nemesis",
                  "type": "nemesis",
                },
              ],
            },
            "4toc": Object {
              "id": "4toc",
              "name": "4 ToC",
              "turnOrderCards": Array [
                Object {
                  "id": "player1-1",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "player1-2",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "player1-3",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "player1-4",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "nemesis-1",
                  "name": "Nemesis",
                  "type": "nemesis",
                },
                Object {
                  "id": "nemesis-2",
                  "name": "Nemesis",
                  "type": "nemesis",
                },
              ],
            },
            "default": Object {
              "id": "default",
              "name": "Default (3 ToC)",
              "turnOrderCards": Array [
                Object {
                  "id": "player1-1",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "player1-2",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "player1-3",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "nemesis-1",
                  "name": "Nemesis",
                  "type": "nemesis",
                },
                Object {
                  "id": "nemesis-2",
                  "name": "Nemesis",
                  "type": "nemesis",
                },
              ],
            },
          },
        },
        "SelectedSetup": Object {
          "id": "default",
          "name": "Default (3 ToC)",
          "turnOrderCards": Array [
            Object {
              "id": "player1-1",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-2",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-3",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "nemesis-1",
              "name": "Nemesis",
              "type": "nemesis",
            },
            Object {
              "id": "nemesis-2",
              "name": "Nemesis",
              "type": "nemesis",
            },
          ],
        },
      },
    ],
    "failActionCreator": [Function],
    "func": [Function],
    "simulate": [Function],
    "successActionCreator": [Function],
    "type": "RUN",
    Symbol(isCmd): true,
  },
]
`;

exports[`Turnorder configuration reducer should handle SET_MODE 1`] = `
Array [
  Object {
    "Mode": "Blitz",
    "SelectedPlayerCount": Object {
      "id": "onePlayer",
      "name": "1",
      "variations": Object {
        "34alternating": Object {
          "id": "34alternating",
          "name": "3/4 Cards alternating",
          "turnOrderCards": Array [
            Object {
              "id": "player1-1",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-2",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-3",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "alternate": true,
              "id": "player-1-alternating",
              "name": "Player 1 (alternate)",
              "type": "player1",
            },
            Object {
              "id": "nemesis-1",
              "name": "Nemesis",
              "type": "nemesis",
            },
            Object {
              "id": "nemesis-2",
              "name": "Nemesis",
              "type": "nemesis",
            },
          ],
        },
        "4toc": Object {
          "id": "4toc",
          "name": "4 ToC",
          "turnOrderCards": Array [
            Object {
              "id": "player1-1",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-2",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-3",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-4",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "nemesis-1",
              "name": "Nemesis",
              "type": "nemesis",
            },
            Object {
              "id": "nemesis-2",
              "name": "Nemesis",
              "type": "nemesis",
            },
          ],
        },
        "default": Object {
          "id": "default",
          "name": "Default (3 ToC)",
          "turnOrderCards": Array [
            Object {
              "id": "player1-1",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-2",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-3",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "nemesis-1",
              "name": "Nemesis",
              "type": "nemesis",
            },
            Object {
              "id": "nemesis-2",
              "name": "Nemesis",
              "type": "nemesis",
            },
          ],
        },
      },
    },
    "SelectedSetup": Object {
      "id": "default",
      "name": "Default (3 ToC)",
      "turnOrderCards": Array [
        Object {
          "id": "player1-1",
          "name": "Player 1",
          "type": "player1",
        },
        Object {
          "id": "player1-2",
          "name": "Player 1",
          "type": "player1",
        },
        Object {
          "id": "player1-3",
          "name": "Player 1",
          "type": "player1",
        },
        Object {
          "id": "nemesis-1",
          "name": "Nemesis",
          "type": "nemesis",
        },
        Object {
          "id": "nemesis-2",
          "name": "Nemesis",
          "type": "nemesis",
        },
      ],
    },
  },
  Object {
    "args": Array [
      "2-4-turnOrderConfiguration",
      Object {
        "Mode": "Blitz",
        "SelectedPlayerCount": Object {
          "id": "onePlayer",
          "name": "1",
          "variations": Object {
            "34alternating": Object {
              "id": "34alternating",
              "name": "3/4 Cards alternating",
              "turnOrderCards": Array [
                Object {
                  "id": "player1-1",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "player1-2",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "player1-3",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "alternate": true,
                  "id": "player-1-alternating",
                  "name": "Player 1 (alternate)",
                  "type": "player1",
                },
                Object {
                  "id": "nemesis-1",
                  "name": "Nemesis",
                  "type": "nemesis",
                },
                Object {
                  "id": "nemesis-2",
                  "name": "Nemesis",
                  "type": "nemesis",
                },
              ],
            },
            "4toc": Object {
              "id": "4toc",
              "name": "4 ToC",
              "turnOrderCards": Array [
                Object {
                  "id": "player1-1",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "player1-2",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "player1-3",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "player1-4",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "nemesis-1",
                  "name": "Nemesis",
                  "type": "nemesis",
                },
                Object {
                  "id": "nemesis-2",
                  "name": "Nemesis",
                  "type": "nemesis",
                },
              ],
            },
            "default": Object {
              "id": "default",
              "name": "Default (3 ToC)",
              "turnOrderCards": Array [
                Object {
                  "id": "player1-1",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "player1-2",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "player1-3",
                  "name": "Player 1",
                  "type": "player1",
                },
                Object {
                  "id": "nemesis-1",
                  "name": "Nemesis",
                  "type": "nemesis",
                },
                Object {
                  "id": "nemesis-2",
                  "name": "Nemesis",
                  "type": "nemesis",
                },
              ],
            },
          },
        },
        "SelectedSetup": Object {
          "id": "default",
          "name": "Default (3 ToC)",
          "turnOrderCards": Array [
            Object {
              "id": "player1-1",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-2",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "player1-3",
              "name": "Player 1",
              "type": "player1",
            },
            Object {
              "id": "nemesis-1",
              "name": "Nemesis",
              "type": "nemesis",
            },
            Object {
              "id": "nemesis-2",
              "name": "Nemesis",
              "type": "nemesis",
            },
          ],
        },
      },
    ],
    "failActionCreator": [Function],
    "func": [Function],
    "simulate": [Function],
    "successActionCreator": [Function],
    "type": "RUN",
    Symbol(isCmd): true,
  },
]
`;

exports[`Turnorder configuration reducer should return the initial state 1`] = `
Object {
  "Mode": "Default",
  "SelectedPlayerCount": Object {
    "id": "onePlayer",
    "name": "1",
    "variations": Object {
      "34alternating": Object {
        "id": "34alternating",
        "name": "3/4 Cards alternating",
        "turnOrderCards": Array [
          Object {
            "id": "player1-1",
            "name": "Player 1",
            "type": "player1",
          },
          Object {
            "id": "player1-2",
            "name": "Player 1",
            "type": "player1",
          },
          Object {
            "id": "player1-3",
            "name": "Player 1",
            "type": "player1",
          },
          Object {
            "alternate": true,
            "id": "player-1-alternating",
            "name": "Player 1 (alternate)",
            "type": "player1",
          },
          Object {
            "id": "nemesis-1",
            "name": "Nemesis",
            "type": "nemesis",
          },
          Object {
            "id": "nemesis-2",
            "name": "Nemesis",
            "type": "nemesis",
          },
        ],
      },
      "4toc": Object {
        "id": "4toc",
        "name": "4 ToC",
        "turnOrderCards": Array [
          Object {
            "id": "player1-1",
            "name": "Player 1",
            "type": "player1",
          },
          Object {
            "id": "player1-2",
            "name": "Player 1",
            "type": "player1",
          },
          Object {
            "id": "player1-3",
            "name": "Player 1",
            "type": "player1",
          },
          Object {
            "id": "player1-4",
            "name": "Player 1",
            "type": "player1",
          },
          Object {
            "id": "nemesis-1",
            "name": "Nemesis",
            "type": "nemesis",
          },
          Object {
            "id": "nemesis-2",
            "name": "Nemesis",
            "type": "nemesis",
          },
        ],
      },
      "default": Object {
        "id": "default",
        "name": "Default (3 ToC)",
        "turnOrderCards": Array [
          Object {
            "id": "player1-1",
            "name": "Player 1",
            "type": "player1",
          },
          Object {
            "id": "player1-2",
            "name": "Player 1",
            "type": "player1",
          },
          Object {
            "id": "player1-3",
            "name": "Player 1",
            "type": "player1",
          },
          Object {
            "id": "nemesis-1",
            "name": "Nemesis",
            "type": "nemesis",
          },
          Object {
            "id": "nemesis-2",
            "name": "Nemesis",
            "type": "nemesis",
          },
        ],
      },
    },
  },
  "SelectedSetup": Object {
    "id": "default",
    "name": "Default (3 ToC)",
    "turnOrderCards": Array [
      Object {
        "id": "player1-1",
        "name": "Player 1",
        "type": "player1",
      },
      Object {
        "id": "player1-2",
        "name": "Player 1",
        "type": "player1",
      },
      Object {
        "id": "player1-3",
        "name": "Player 1",
        "type": "player1",
      },
      Object {
        "id": "nemesis-1",
        "name": "Nemesis",
        "type": "nemesis",
      },
      Object {
        "id": "nemesis-2",
        "name": "Nemesis",
        "type": "nemesis",
      },
    ],
  },
}
`;
